extensions/src/org/pentaho/platform/plugin/services/security/userrole/jdbc/JdbcUserRoleListService.java

```diff
@@ -241,6 +256,13 @@ protected Object mapRow( final ResultSet rs, final int rownum ) throws SQLExcept
       }
     }
 
+    // Now add extra role if it does not exist in the list
+    for ( String extraRole : extraRoles ) {
+      if ( !roles.contains( extraRole ) ) {
```

Comment:
As before, this will be costly with a lot of roles. No need to do a contains or similar if you just use a LinkedHashSet